model/daal4py-LR/daal4py_LR/daal4py.py

```diff
@@ -0,0 +1,95 @@
+import pathlib
+import statistics
+import importlib
+from typing import AsyncIterator, Tuple, Any, Type, List
+
+from dffml import (
+    config,
+    field,
+    entrypoint,
+    SimpleModel,
+    ModelNotTrained,
+    Accuracy,
+    Feature,
+    Features,
+    Sources,
+    Record,
+)
+
+
+@config
+class daal4pyModelConfig:
+    predict: Feature = field("Label or the value to be predicted")
+    features: Features = field("Features to train on. For SLR only 1 allowed")
+    directory: pathlib.Path = field(
+        "Directory where state should be saved",
+        default=pathlib.Path("~", ".cache", "dffml", "daal4py"),
+    )
+
+class daal4pyModelContext(SimpleModelContext):
+    """
+    Model using tensorflow to make predictions. Handels creation of feature
+    columns for real valued, string, and list of real valued features.
+    """
+
+    def __init__(self, parent) -> None:
+        super().__init__(parent)
+        self.d4p = importlib.import_module("daal4py")
+        self.pd = importlib.import_module("pandas")
+        self.np = importlib.import_module("numpy")
+        self.features = self.applicable_features
+        self.lm = self.d4p.linear_regression_training(interceptFlag=True)
+        self.lm_trained = None
+        self.lm_predictor = None
+
+    async def train(self, sources: Sources) -> None:
+        data = []
+        async for record in sources.with_features(
+            self.features + [self.parent.config.predict.NAME]
+        ):
+            feature_data = record.features(
+                self.features + [self.parent.config.predict.NAME]
+            )
+            data.append(feature_data)
+        df = self.pd.DataFrame(data)
+        xdata = self.np.array(df.drop([self.parent.config.predict.NAME], 1))
+        ydata = self.np.array(df[self.parent.config.predict.NAME])
+        self.lm_trained = self.lm.compute(xdata, ydata)
```

Comment:
Those streaming docs seemed to suggest to could call the `compute` function multiple times. That would eliminate the need to store data in memory. We could just call it in the body of the `for` loop

Reply:
Also, when we're done with the `for` loop, we need to figure out how to save the model state to disk

Reply:
Ah, I see. Thank you.

Reply:
> Also, when we're done with the `for` loop, we need to figure out how to save the model state to disk

Sorry, I don't follow.